components/scream/src/physics/shoc/tests/shoc_grid_tests.cpp

```diff
@@ -0,0 +1,135 @@
+#include "catch2/catch.hpp"
+
+//#include "share/scream_types.hpp"
+#include "ekat/scream_types.hpp"
```

Comment:
@pressel I've been focusing on this file.  These tests aren't exactly what I had in mind, but i'm actually glad because it gave me a chance to check this out and give myself a c++/testing crash course.  I have modifications that more reflect my vision for this that run/pass.  Did you want me to commit changes to this branch or hold off and make a separate PR once this goes through?

Reply:
@bogensch I'm glad you are looking at the tests and have been thinking about extending them. The tests that are implemented are very simple tests to test that the bridge function is working (hence the WIP PR). I do have your specific tests coded up in separate local dev branch and my plan was to go back and add the specific tests you mentioned once we have agreed that the design patterns used are those that we will be sticking with.  That said, if you would like to contribute your tests that is fine with me.  Since @jgfouca has signed off on the design I think we are now at that point.  

Let me finish responding to @jgfouca's review which requires some substantial modifications to the this file, and then we can decide if committing to the branch or a separate PR is best.

Reply:
Update: @bogensch I just pushed a new commit that along with a fair amount of cleanup should add the tests we had talked about previously. Let me know if these weren't the ones you had in mind and feel free to add or request others.

Reply:
Thanks @pressel , yes the tests are definitely more what I imagined.  A question about how the input data is defined... Is it necessary that we do these tests with the 128 full model levels?  When I made modifications to the tests I was constructing a simple five layer model and explicitly defining the zt and zi grids (allowing for stretched grids to be tested) as well as the density varying with height as such (see screenshot).  This is more how I envisioned input data would be defined for most SHOC tests.  

<img width="632" alt="Screen Shot 2020-06-30 at 11 08 40 AM" src="https://user-images.githubusercontent.com/25110300/86162266-625ef200-bac3-11ea-925c-d00ffcb70a83.png">

Reply:
@bogensch Yes, we can definitely specify the input this way.

Reply:
@bogensch, I implemented the test you mention above. Let me know if you want me to add any others.